File: DotNet/Kyu8/GenerateRangeOfIntegers.cs
// https://www.codewars.com/kata/generate-range-of-integers/train/csharp

using System;

namespace DotNet.Kyu8;

using System.Collections.Generic;

public class GenerateRangeOfIntegers
{
    public static int[] GenerateRange(int min, int max, int step)
    {
        List<int> list = new List<int>();

        while (min <= max)
        {
            list.Add(min);
            min += step;
        }

        return list.ToArray();
    }

    public static void Main()
    {
        Console.WriteLine(string.Join(", ", GenerateRange(1, 10, 1)));
        Console.WriteLine(string.Join(", ", GenerateRange(-10, 1, 1)));
        Console.WriteLine(string.Join(", ", GenerateRange(1, 15, 20)));
    }
}

File: Elixir/Kyu8/generate-range-of-integers.ex
# https://www.codewars.com/kata/generate-range-of-integers/train/elixir

defmodule NumGenerator do
  defp element_at_index(index, min, step), do: min + index * step

  def generate_range(min, max, step) do
    count = Integer.floor_div(max - min, step)
    for index <- 0..count, do: element_at_index(index, min, step)
  end

  # :lists.seq(min, max, step)
  # Stream.take_every(min..max, step) |> Enum.to_list()
end

IO.inspect(NumGenerator.generate_range(1, 10, 4))
